### PR TITLE
feat: support login UI customization

### PR DESCRIPTION
**Pull request by 1340908470**
Fix: https://github.com/casbin/casdoor/issues/142

### CODE REVIEW

**Comment by hsluoyz** [comment]
@1340908470 demo

**Comment by hsluoyz** [comment]
@Kininaru @sh1luo plz review.

File: web/src/auth/LoginPageCustom.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                    Setting.goToLink("/");
+                } else {
+                    Util.showMessage("error", `Failed to log in: ${res.msg}`);
+                }
+            });
+    };
+
+    componentDidMount() {
+        console.log(this.props)
+        if (document.getElementById("login") && document.getElementById("username") &&
```
Comment (sh1luo) [inline web/src/auth/LoginPageCustom.js:31]: Do not show debugging info.

File: web/src/auth/LoginPage.js [hunk 2]
```diff
@@ -376,26 +377,32 @@ class LoginPage extends React.Component {
...
-              this.renderForm(application)
-            }
-          </div>
-        </Col>
-      </Row>
-    )
+      if ((this.state.application && this.state.application.loginPageCustom !== "")) {
+          return <LoginPageCustom Parent={this} Application={this.state.application} Html={this.state.application.loginPageCustom}/>
+      } else if (this.state.classes.application && this.state.classes.application.loginPageCustom !== "") {
+          return <LoginPageCustom Parent={this} Application={this.state.classes.application } Html={this.state.classes.application.loginPageCustom}/>
```
Comment (sh1luo) [inline web/src/auth/LoginPage.js:383]: `if (this.state.application?.loginPageCustom  !== "")`, other than `app&&app.loginPageCustom`

File: yarn.lock [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
```
Comment (sh1luo) [inline yarn.lock:4]: Does not seem to need this lock file

**Review by sh1luo — CHANGES_REQUESTED** [review]

File: web/src/auth/LoginPageCustom.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import React from 'react'
```
Comment (Kininaru) [inline web/src/auth/LoginPageCustom.js:1]: Casbin License

**Review by Kininaru — COMMENTED** [review]
Please do not use DOM APIs. Use React state instead.

**Review by Kininaru — APPROVED** [review]

**Comment by hsluoyz** [comment]
@1340908470 plz remember to provide demo link for each PR.

**Comment by 1340908470** [comment]
[Demo Link](http://chenweihao.vip/)

### test html string:
```html
<!DOCTYPE html>
<html>

<head>
    <meta charset="UTF-8">
    <title>登录</title>
    <link rel="stylesheet" href="../css/login.css" />
    <link rel="stylesheet" href="https://cdn.staticfile.org/font-awesome/4.7.0/css/font-awesome.css">
    <style>
        body {
            background: url('../image/2.jpg');
            background-repeat: no-repeat;
            background-size: 100% auto;
        }

        #login-box {
            width: 70%;
            height: auto;
            margin: 0 auto;
            margin-top: 13%;
            text-align: center;
            background: #00000060;
            padding: 20px 50px;
        }

        #login-box h1 {
            color: #fff;
        }

        #login-box .form .item {
            margin-top: 15px;
        }

        #login-box .form .item i {
            font-size: 18px;
            color: #fff;
        }

        #login-box .form .item input {
            width: 180px;
            font-size: 18px;
            border: 0;
            border-bottom: 2px solid #fff;
            padding: 5px 10px;
            background: #ffffff00;
            color: #fff;
        }

        #login-box button {
            margin-top: 20px;
            width: 190px;
            height: 30px;
            font-size: 20px;
            font-weight: 700;
            color: #fff;
            background-image: linear-gradient(to right, #74ebd5 0%, #9face6 100%);
            border: 0;
            border-radius: 15px;
        }
    </style>
</head>

<body>
    <div id="login-box">
        <h1>Login</h1>
        <div class="form">
            <div class="item"> <i class="fa fa-github-alt" style="font-size:24px"></i> <input id="username" type="text"
                    placeholder="Username"></div>
            <div class="item">
                <i class="fa fa-search" style="font-size:24px"></i> <input id="password" type="password"
                    placeholder="Password">
            </div>
            <input id="remember" name="checkbox" type="checkbox" value="checkbox" checked="checked" />
            Auto login
        </div> <button id="login">Login</button>
        <br/>
        <a id="forget">forgot password</a>
        <br/>
        <a id="signup">sign up now</a>
    </div>
</body>

</html>
```

**Comment by hsluoyz** [comment]
@1340908470 plz resolve:

image

**Comment by hsluoyz** [comment]
@1340908470 plz fix the wrong commits.

**Pull request closed without merging**